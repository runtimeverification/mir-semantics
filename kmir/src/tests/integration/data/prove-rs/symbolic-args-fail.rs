#[allow(dead_code)]
struct MyStruct<'a>{ field: &'a MyEnum}

#[allow(dead_code)]
enum MyEnum {
    My1,
    My2(i8),
}

#[allow(dead_code)]
#[allow(unused)]
fn eats_all_args<'a>(
    x1: i32, 
    x2: &mut u16,
    x3: bool,
    x4: MyStruct<'a>,
    x5: MyEnum,
    x6: &mut [u8],
    x7: &[i8; 3],
    x8: &mut [MyStruct<'a>; 2],
    x9: *const MyStruct<'a>,
) -> () {
    *x2 = x1 as u16;
    if x3 { 
        x8[0] = x4;
    }
    else {
        x8[0] = unsafe { MyStruct{field: (*x9).field} };
    }
    match x5 {
        _ => { 
            if x6.len() > 0
                { x6[0] = x7[0] as u8; }
        }
    }
}

// we need a `main` function that calls eats_all_args
fn main() {
    let e1 = MyEnum::My1;
    let e2 = MyEnum::My2(0);
    let e3 = MyEnum::My1;
    let e4 = MyEnum::My2(0);
    let mut x2 = 0;
    let my1 = MyStruct{field: &e1};
    let my2 = MyStruct{field: &e2};
    let my3 = MyStruct{field: &e3};
    let my4 = MyStruct{field: &MyEnum::My1};
    let mut a1 = [1, 2, 3];
    let a2 = [1, 2, 3];
    eats_all_args(1, &mut x2, true, my1, e4, &mut a1, &a2, &mut [my2, my3], &my4 as *const MyStruct<'_>);

    assert!(false); // makes the test with main fail, as the other one also fails
}
