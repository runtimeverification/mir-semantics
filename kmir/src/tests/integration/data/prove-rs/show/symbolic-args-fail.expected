
┌─ 1 (root, init)
│   #execTerminator ( terminator ( ... kind: terminatorKindCall ( ... func: operandC
│   span: 0
│
│  (45 steps)
├─ 3 (split)
│   #selectBlock ( switchTargets ( ... branches: branch ( 0 , basicBlockIdx ( 3 ) )
┃
┃ (branch)
┣━━┓ subst: .Subst
┃  ┃ constraint:
┃  ┃     notBool ARG_BOOL3:Bool
┃  │
┃  ├─ 4
┃  │   #selectBlock ( switchTargets ( ... branches: branch ( 0 , basicBlockIdx ( 3 ) )
┃  │
┃  │  (37 steps)
┃  ├─ 6
┃  │   #selectBlock ( switchTargets ( ... branches: branch ( 0 , basicBlockIdx ( 7 ) )
┃  ┃
┃  ┃ (1 step)
┃  ┣━━┓
┃  ┃  │
┃  ┃  ├─ 8
┃  ┃  │   #execBlockIdx ( basicBlockIdx ( 7 ) ) ~> .K
┃  ┃  │
┃  ┃  │  (4 steps)
┃  ┃  ├─ 14 (terminal)
┃  ┃  │   #EndProgram ~> .K
┃  ┃  │
┃  ┃  ┊  constraint: true
┃  ┃  ┊  subst: ...
┃  ┃  └─ 2 (leaf, target, terminal)
┃  ┃      #EndProgram ~> .K
┃  ┃
┃  ┣━━┓
┃  ┃  │
┃  ┃  ├─ 9
┃  ┃  │   #selectBlock ( switchTargets ( ... branches: .Branches , otherwise: basicBlockId
┃  ┃  │
┃  ┃  │  (85 steps)
┃  ┃  └─ 15 (vacuous, leaf)
┃  ┃      #expect ( typedValue ( BoolVal ( 0 <Int size ( ARG_ARRAY7:List ) ) , ty ( 28 ) ,
┃  ┃
┃  ┗━━┓
┃     │
┃     └─ 10 (stuck, leaf)
┃         #selectBlock ( switchTargets ( ... branches: branch ( 0 , basicBlockIdx ( 7 ) )
┃
┗━━┓ subst: .Subst
   ┃ constraint:
   ┃     ARG_BOOL3:Bool
   │
   ├─ 5
   │   #selectBlock ( switchTargets ( ... branches: branch ( 0 , basicBlockIdx ( 3 ) )
   │
   │  (87 steps)
   ├─ 7
   │   #selectBlock ( switchTargets ( ... branches: branch ( 0 , basicBlockIdx ( 7 ) )
   ┃
   ┃ (1 step)
   ┣━━┓
   ┃  │
   ┃  ├─ 11
   ┃  │   #execBlockIdx ( basicBlockIdx ( 7 ) ) ~> .K
   ┃  │
   ┃  │  (4 steps)
   ┃  ├─ 16 (terminal)
   ┃  │   #EndProgram ~> .K
   ┃  │
   ┃  ┊  constraint: true
   ┃  ┊  subst: ...
   ┃  └─ 2 (leaf, target, terminal)
   ┃      #EndProgram ~> .K
   ┃
   ┣━━┓
   ┃  │
   ┃  ├─ 12
   ┃  │   #selectBlock ( switchTargets ( ... branches: .Branches , otherwise: basicBlockId
   ┃  │
   ┃  │  (85 steps)
   ┃  └─ 17 (vacuous, leaf)
   ┃      #expect ( typedValue ( BoolVal ( 0 <Int size ( ARG_ARRAY7:List ) ) , ty ( 28 ) ,
   ┃
   ┗━━┓
      │
      └─ 13 (stuck, leaf)
          #selectBlock ( switchTargets ( ... branches: branch ( 0 , basicBlockIdx ( 7 ) )



